### PR TITLE
fixing file close bug

### PR DESCRIPTION
**Pull request by vipragupta**
* Fixed that the file is not closed after writing to the file.
* Cleaning the adhoc tasks.

### CODE REVIEW

File: utils/file/file.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func OverwriteFile(file string, data []byte) {
 	if err != nil {
 		log.Errorln("Error creating file")
 	}
```
Comment (sjlee) [inline utils/file/file.go:264]: The change to close the file at all times looks fine. But here it seems that if we fail to create the file we still seem to continue on and try to write to the non-existent file. We should return early if we fail to create the file, no?

File: utils/file/file.go [hunk 2]
```diff
@@ -252,20 +252,23 @@ func DumpPluginModifiedComposeFiles(ctx context.Context, plugin, funcName string
...
 }
 
-func OverwriteFile(file string, data []byte) {
+func OverwriteFile(file string, data []byte) error {
```
Comment (sjlee) [inline utils/file/file.go:255]: Now that `OverwriteFile` returns an error, all callers of this function should check for the error and handle it appropriately. Could you please look into its callers and make corresponding changes?

Comment (vipragupta) [reply utils/file/file.go:255]: yes. I verified that

Comment (sjlee) [reply utils/file/file.go:255]: Interesting that there doesn't seem to be any caller of this function.

**Pull request merged**